### PR TITLE
Add FreeBSD support, fix various things

### LINKED ISSUE #41: Temperature not reported correctly with EddyStone TLM frames.
It looks like there is a signed vs unsigned issue?  I have a beacon inside a cooler with dry ice.  The temperature inside is about -36 degrees C.  My script is reporting 220 degrees C.  If you interpret this number as a two's complement you get the right answer.  Here is what is reported:

<ac:23:3f:2a:e0:4e, -49> EddystoneTLMFrame<voltage: 3216 mV, temperature: 220 Celsius, advertising count: 12513900, seconds since boot: 99146930>

Here is my code:

import time

from beacontools import BeaconScanner, IBeaconFilter

def callback(bt_addr, rssi, packet, additional_info):
    str = "<%s, %d> %s %s" % (bt_addr, rssi, packet, time.time())
    print(str)

# scan for all advertisements from beacons
scanner = BeaconScanner(callback)
scanner.start()
time.sleep(4)
scanner.stop()

### LINKED ISSUE #43: Example Scanner working on Raspberry Pi, but not on Debian Buster Linux Laptop.
The program works almost perfectly on the Raspberry Pi, but the exact same program does not return any thing (even the devices the Raspberry Pi detects) on my Lenovo Thinkpad X1 Carbon (running Debian Buster). sudo hcitools lescan returns "Set scan parameters failed: Input/output error". Nothing I do from the countless online suggestions seems to make that error go away. I've restarted hci0, restarted my machine, restarted bluetooth services, etc and the error still persists. bluetoothctl however can scan and detect the le iBeacon messages. Can someone help me figure this out?

### PR DESCRIPTION
**Pull request by myfreeweb**
- Split BlueZ backend into a separate module (hopefully correctly, please test)
- Add FreeBSD backend
- Fix #41
- Hopefully fix #43 (by not using randomized MAC)

### CODE REVIEW

File: beacontools/backend/__init__.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import sys
+
+if sys.platform.startswith("linux"):
+    from .bluez import *
+elif sys.platform.startswith("freebsd"):
+    from .freebsd import *
+else:
+    raise NotImplementedError("Scanning not supported on this platform")
```
Comment (citruz) [inline beacontools/backend/__init__.py:8]: With this you are effectively removing support for Windows and macOS which are also supported by the pybluez backend. Please change this so that only on FreeBSD that backend is used and bluez otherwise.
```suggestion
if sys.platform.startswith("freebsd"):
    from .freebsd import *
else:
    from .bluez import *
```

File: beacontools/packet_types/eddystone.py [hunk 2]
```diff
@@ -98,8 +98,8 @@ class EddystoneTLMFrame(object):
 
     def __init__(self, data):
         self._voltage = data['voltage']
-        self._temperature = data['temperature']
-        self._temperature_fixed_point = data['temperature'] / float(256)
```
Comment (citruz) [inline beacontools/packet_types/eddystone.py:102]: What is the reason behind this change?

File: beacontools/scanner.py [hunk 3]
```diff
@@ -108,7 +103,7 @@ def run(self):
         self.socket.close()
 
     def set_scan_parameters(self, scan_type=ScanType.ACTIVE, interval_ms=10, window_ms=10,
-                            address_type=BluetoothAddressType.RANDOM, filter_type=ScanFilter.ALL):
+                            address_type=BluetoothAddressType.PUBLIC, filter_type=ScanFilter.ALL):
```
Comment (citruz) [inline beacontools/scanner.py:106]: I don't like changing this globally because it has privacy implications. Scanning should be done using randomized MACs to avoid the possibility of tracking. If this really fixes #43 we should rather add a parameter to `BeaconScanner.__init__` so that users who need this option can set it.

File: beacontools/scanner.py [hunk 4]
```diff
@@ -141,14 +136,14 @@ def set_scan_parameters(self, scan_type=ScanType.ACTIVE, interval_ms=10, window_
         interval_fractions, window_fractions = int(interval_fractions), int(window_fractions)
 
         scan_parameter_pkg = struct.pack(
-            ">BHHBB",
+            "<BHHBB",
```
Comment (citruz) [inline beacontools/scanner.py:139]: why did you change this?

File: beacontools/scanner.py [hunk 5]
```diff
@@ -157,8 +152,8 @@ def toggle_scan(self, enable, filter_duplicates=False):
             enable: boolean value to enable (True) or disable (False) scanner
             filter_duplicates: boolean value to enable/disable filter, that
                 omits duplicated packets"""
-        command = struct.pack(">BB", enable, filter_duplicates)
-        self.bluez.hci_send_cmd(self.socket, OGF_LE_CTL, OCF_LE_SET_SCAN_ENABLE, command)
+        command = struct.pack("BB", enable, filter_duplicates)
```
Comment (citruz) [inline beacontools/scanner.py:155]: why did you change this?

File: setup.py [hunk 6]
```diff
@@ -63,7 +64,7 @@
     # for example:
     # $ pip install -e .[dev,test]
     extras_require={
-        'scan': ['PyBluez==0.22'],
+        'scan': ['PyBluez==0.22'] if sys.platform.startswith("linux") else [],
```
Comment (citruz) [inline setup.py:67]: See first comment. Please only add special handling for FreeBSD and leave it otherwise as is.

File: beacontools/packet_types/eddystone.py (hunk 2 above)
Comment (citruz) [reply beacontools/packet_types/eddystone.py:102]: Aaah I see. I parsed the temperature wrong. It must big-endian and then divided by 256, which results in the exact same value if you have no decimals like the beacons I am using🤦‍♂️

Comment (citruz) [reply beacontools/packet_types/eddystone.py:102]: You can remove `_temperature_fixed_point` attribute completely then.

File: beacontools/backend/__init__.py (hunk 1 above)
Comment (myfreeweb) [reply beacontools/backend/__init__.py:8]: No, they were never supported!! (i wrote this in the commit message)

The `bluetooth._bluetooth` module that this code imports is literally the [C extension that binds to bluez libs](https://github.com/karulis/pybluez/blob/06f3c83dcd7857c7f97aa1f882357725fc4342d4/setup.py#L80-L84).

PyBluez mentions Windows and Mac in the README because it has some generic Bluetooth Classic support on all three platforms. That's not what you're using.

File: beacontools/scanner.py (hunk 4 above)
Comment (myfreeweb) [reply beacontools/scanner.py:139]: Because Bluetooth uses little endian everywhere (again, see commit messages)

How did you get the idea that this should be big endian?

File: beacontools/scanner.py (hunk 5 above)
Comment (myfreeweb) [reply beacontools/scanner.py:155]: Here, you do not need any endianness specifier, as single byte values have no endianness

File: beacontools/backend/__init__.py (hunk 1 above)
Comment (citruz) [reply beacontools/backend/__init__.py:8]: Ah ok, I never tried it and thus didn't realize that it doesn't work in these platforms.

File: beacontools/scanner.py (hunk 4 above)
Comment (citruz) [reply beacontools/scanner.py:139]: I never questioned it because I thought it works™. But now I realize changing the interval and window values has no impact if they are big endian. Thanks for the hint!

**Comment by citruz** [comment]
I will merge the PR and add the scan parameter myself. Thanks for the contribution!
